### PR TITLE
Commented and Refactored UART2.h and UART2.c to remove unused functions

### PR DESCRIPTION
**Pull request by sergei1152**
Cleaned up and documented UART2.c and UART2.h and removed instances of PATH_MANAGER and GPS_OLD from the files, as well as removed `UART2_SendChar` and `UART2_SendString` functions since they werent being used anywhere

### CODE REVIEW

File: Autopilot/AttitudeManager/UART2.c [hunk 1]
```diff
@@ -61,22 +66,14 @@ void InitUART2()
...
+  IPC7 = 0x4400;	// Mid Range Interrupt Priority level, no urgent reason
 
 	IFS1bits.U2TXIF = 0;	// Clear the Transmit Interrupt Flag
 	IEC1bits.U2TXIE = 0;	// Enable Transmit Interrupts
 	IFS1bits.U2RXIF = 0;	// Clear the Recieve Interrupt Flag
 
-#if PATH_MANAGER && !GPS_OLD
-	IEC1bits.U2RXIE = 1;	// Enable Recieve Interrupts
-#else
-        IEC1bits.U2RXIE = 0;	// Enable Recieve Interrupts
```
Comment (chrishajduk84) [inline Autopilot/AttitudeManager/UART2.c]: Double check that this doesn't break anything. The else statement could be troublesome. 

If I'm not mistaken UART 2 is used for the debug cable....So we don't need a receive interrupt. Double check to be sure.

File: Autopilot/AttitudeManager/main.h [hunk 2]
```diff
@@ -23,6 +23,7 @@
 //                  heading for the aircraft to follow.
 //  Attitude Manager - Communicates with a IMU (kinematics sensor) in order to
 //                     the desires Pitch, Roll, Yaw on the aircraft.
+//TODO References to these should be stripped out, as they are now in seperate code bases - Serge Sept 2016
```
Comment (chrishajduk84) [inline Autopilot/AttitudeManager/main.h]: MPLab has a find in projects search feature.

File: Autopilot/AttitudeManager/UART2.h [hunk 3]
```diff
@@ -1,35 +1,18 @@
...
+ * @author Mitch Hatfield
+ * @date Jun 2013
+ * @brief Settings for UART2 communication used for telemetry link 
+ * @copyright Waterloo Aerial Robotics Group 2016 \n
+ *   https://raw.githubusercontent.com/UWARG/PICpilot/master/LICENCE 
  */
 
-#ifndef UART2_H
-#define	UART2_H
+#pragma once
```
Comment (chrishajduk84) [inline Autopilot/AttitudeManager/UART2.h]: Does this actually work?

File: Autopilot/AttitudeManager/UART2.h [hunk 4]
```diff
@@ -1,35 +1,18 @@
...
+
 #include "main.h"
 
+/** Initializes the UART2 connection **/
 void InitUART2();
-void UART2_SendString(char *s);
-void UART2_SendChar(char data);
```
Comment (chrishajduk84) [inline Autopilot/AttitudeManager/UART2.h]: This will break it. The send string and send char are still used in other parts of the code.

Comment (sergei1152) [inline Autopilot/AttitudeManager/UART2.h]: I did a find usage with the IDE and the only time its used is in the implementation

File: Autopilot/AttitudeManager/UART2.h (hunk 3 above)
Comment (sergei1152) [inline Autopilot/AttitudeManager/UART2.h]: It compiled and built the project so I'm assuming so?

File: Autopilot/AttitudeManager/main.h (hunk 2 above)
Comment (sergei1152) [inline Autopilot/AttitudeManager/main.h]: Yes, I put that into another task however

File: Autopilot/AttitudeManager/UART2.c (hunk 1 above)
Comment (sergei1152) [inline Autopilot/AttitudeManager/UART2.c]: UART2 is used for the Xbee as documented here http://docs.uwarg.com/picpilot/uart/

Comment (chrishajduk84) [inline Autopilot/AttitudeManager/UART2.c]: Okay... But if we disable receive interrupts,  won't that break it?

File: Autopilot/AttitudeManager/UART2.h (hunk 4 above)
Comment (chrishajduk84) [inline Autopilot/AttitudeManager/UART2.h]: Double check debug.c.

When you enable Debugging,  it should use these functions to spit stuff out..... . Actually that's UART1. You are right. I think this is okay.

File: Autopilot/AttitudeManager/UART2.c (hunk 1 above)
Comment (sergei1152) [inline Autopilot/AttitudeManager/UART2.c]: I'm not sure, however the code hasn't changed from what was there before. The preprocessor would have just pasted in `IEC1bits.U2RXIE = 0;`, so this change should affect any of the exisitng functionality

Comment (chrishajduk84) [inline Autopilot/AttitudeManager/UART2.c]: If you are removing IEC1bits.U2RXIE = 0; it won't paste that in anymore though...

Comment (sergei1152) [inline Autopilot/AttitudeManager/UART2.c]: I havent removed it. I moved it to be outside the if else. Check the file diff

**Pull request merged**